packages/blade/src/components/Typography/BaseText/BaseText.tsx

```diff
@@ -0,0 +1,128 @@
+import type { ReactElement } from 'react';
+import getIn from '../../../utils/getIn';
+import { useTheme } from '../../BladeProvider';
+import isEmpty from '../../../utils/isEmpty';
+import makeTypographySize from '../../../utils/makeTypographySize';
+import type { ColorContrast } from '../../../tokens/theme/theme';
+import type { Theme } from '../../BladeProvider';
+import StyledBaseText from './StyledBaseText';
+
+/**
+ * @template TokenType token type generic
+ * @description Tokenises objects to dot notation strings, eg: `surface.text.normal.lowContrast`
+ */
+type DotNotationStringToken<TokenType> = {
+  [K in keyof TokenType]: `${Extract<K, number | string>}.${TokenType[K] extends Record<
+    string,
+    ColorContrast | string
+  >
+    ? Extract<keyof TokenType[K], number | string>
+    : DotNotationStringToken<TokenType[K]>}`;
+}[keyof TokenType];
+
+type FeedbackColors = `feedback.text.${DotNotationStringToken<
+  Theme['colors']['feedback']['text']
+>}`;
+type SurfaceColors = `surface.text.${DotNotationStringToken<Theme['colors']['surface']['text']>}`;
+type ActionColors = `action.text.${DotNotationStringToken<Theme['colors']['action']['text']>}`;
+
+export type BaseTextProps = {
+  color: ActionColors | FeedbackColors | SurfaceColors;
+  fontFamily: keyof Theme['typography']['fonts']['family'];
+  fontSize: keyof Theme['typography']['fonts']['size'];
+  fontWeight: keyof Theme['typography']['fonts']['weight'];
+  fontStyle?: 'italic' | 'normal';
+  textDecorationLine?: 'line-through' | 'none';
+  lineHeight: keyof Theme['typography']['lineHeights'];
+  as?: 'code' | 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6' | 'p' | 'span';
+  textAlign?: 'center' | 'justify' | 'left' | 'right';
+  children?: React.ReactNode | undefined;
+};
+
+export const getInvalidPropValueError = ({
```

Comment:
extract this into utils and accept another param for `component`?

Reply:
yes that's the plan, don't want all the abstractions in one shot